resources/nspawn.go

```diff
@@ -0,0 +1,308 @@
+		if send {
+			startup = true // startup finished
+			send = false
+			if exit, err := obj.DoSend(processChan, ""); exit || err != nil {
+				return err // we exit or bubble up a NACK
+			}
+		}
+	}
+}
+
+// CheckApply is run to check the state and, if apply is true, to apply the
+// necessary changes to reach the desired state. this is run before Watch and
+// again if watch finds a change occurring to the state
+func (obj *NspawnRes) CheckApply(apply bool) (checkok bool, err error) {
+	log.Printf("%v[%v]: CheckApply(%t)", obj.Kind(), obj.GetName(), apply)
+
+	if obj.isStateOK { // cache the state
+		return true, nil
+	}
+
+	// this resource depends on systemd ensure that it's running
+	if !systemdUtil.IsRunningSystemd() {
+		return false, errors.New("Systemd is not running.")
+	}
+
+	// connect to org.freedesktop.machine1.Manager
+	conn, err := machined.New()
+	if err != nil {
+		return false, fmt.Errorf("Failed to connect to dbus: %s", err)
+	}
+
+	// compare the current state with the desired state and perform the
+	// appropriate action
+	var exists = true
+	properties, err := conn.GetProperties(obj.GetName())
+	if err != nil {
+		if err, ok := err.(dbus.Error); ok && err.Name !=
+			"org.freedesktop.machine1.NoSuchMachine" {
+			return false, err
+		}
+		exists = false
+		// if we could not successfully get the properties because
+		// there's no such machine the machine is stopped
+		// error if we need the image ignore if we don't
+		if _, err = conn.GetImage(obj.GetName()); err != nil && obj.State != stopped {
+			return false, fmt.Errorf(
+				"No machine nor image named '%s'",
+				obj.GetName())
+		}
+	}
+
+	// if the machine doesn't exist and is supposed to be stopped or
+	// the state matches we're done
+	if !exists && obj.State == stopped || properties["State"] == obj.State {
+		return true, nil
+	}
+
+	// end of state checking. if we're here, checkok is false
+	if !apply {
+		return false, nil
+	}
+
+	if obj.State == running {
+		// start the machine using svc resource
+		return obj.svc.CheckApply(apply)
+
+	}
+	if obj.State == stopped {
+		// terminate the machine with
+		// org.freedesktop.machine1.Manager.KillMachine
+		log.Printf("Stopping nspawn machine: %s", obj.GetName())
+		if err := conn.KillMachine(obj.GetName()); err != nil {
+			return false, err
+		}
+	}
+	return false, nil
+}
+
+// NspawnUID is a unique resource identifier
+type NspawnUID struct {
+	// NOTE: there is also a name variable in the BaseUID struct, this is
+	// information about where this UID came from, and is unrelated to the
+	// information about the resource we're matching. That data which is
+	// used in the IFF function, is what you see in the struct fields here
+	BaseUID
+	name string // the machine name
+}
+
+// IFF aka if and only if they are equivalent, return true. If not, false
+func (obj *NspawnUID) IFF(uid ResUID) bool {
+	res, ok := uid.(*NspawnUID)
+	if !ok {
+		return false
+	}
```

Comment:
There's no access to the NspawnRes object here, so no access to svc.

If I'm reading this right, https://github.com/purpleidea/mgmt/blob/master/resources/resources.go#L158-L166 , we don't need to nest it because the fact that GetUIDs returns both it's own and the one for the svc resource means that both will be checked with their own IFF calls.